src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -23,5 +24,32 @@ public static class MemoryMarshal
         /// </remarks>
         public static Memory<T> AsMemory<T>(ReadOnlyMemory<T> readOnlyMemory) =>
             Unsafe.As<ReadOnlyMemory<T>, Memory<T>>(ref readOnlyMemory);
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ref T GetReference<T>(in Span<T> span) => ref span.Reference;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ref readonly T GetReference<T>(in ReadOnlySpan<T> span) => ref span.Reference;
+
+        public static bool TryGetArray<T>(ReadOnlyMemory<T> readOnlyMemory, out ArraySegment<T> arraySegment)
+        {
+            object obj = readOnlyMemory.GetObjectStartLength(out int index, out int length);
+            if (index < 0)
+            {
+                if (((OwnedMemory<T>)obj).TryGetArray(out var segment))
```

Comment:
Can this be just `return (((OwnedMemory<T>)obj).TryGetArray(out var arraySegment)` ?

Reply:
I see - you need to add the local index.